### PR TITLE
EM-93: Updated Jenkinsfile.dev to include Sonarqube scan for project.

### PR DESCRIPTION
**Pull request by esune**
Changes include:
- adding the BDDStack step to have SonarQube scans every time a change is pushed to Dev.
- minor tweaks to the Jenkinsfile: there apparently aren't conventions on how to structure the Jenkinsfile, I simplified a bit the Dev pipeline in order to make it more straightforward (no need to have arrays of variables if there only is one option).

### CODE REVIEW

File: Jenkinsfile.dev [hunk 1]
```diff
@@ -67,43 +53,69 @@ if(hasRepoChanged){
               type: "button",
               text: "View Build Logs",
               style:"danger",           
-              url: "${currentBuild.absoluteUrl}/console"
+              APP_URL: "${currentBuild.absoluteAPP_URL}/console"
```
Comment (asanchezr) [inline Jenkinsfile.dev]: Question: Is `${currentBuild.absoluteAPP_URL}` correct? It got changed from `${currentBuild.absoluteUrl}` and seems like a text replacement for "url" gone wrong...

File: Jenkinsfile.dev [hunk 2]
```diff
@@ -4,49 +4,35 @@ library identifier: 'devops-library@master', retriever: modernSCM(
 
 // Edit your app's name below
 def APP_NAME = 'esm-server-dev'
-def APP_NAME_URL = 'esm' // this is necessary until we consolidate the naming conventions in OpenShift
-// Edit your environment TAG names below
-def TAG_NAMES = [
-  'dev'
-]
+def APP_NAME_APP_URL = 'esm' // this is necessary until we consolidate the naming conventions in OpenShift
```
Comment (asanchezr) [inline Jenkinsfile.dev]: Same here... should be `APP_NAME_URL`

File: Jenkinsfile.dev [hunk 3]
```diff
@@ -4,49 +4,35 @@ library identifier: 'devops-library@master', retriever: modernSCM(
...
+
 // TODO: fix once OpenShift conventions are in place
-def APP_URLS = [
-  "https://${APP_NAME_URL}-master.${env.PATHFINDER_URL}/"
-]
+def APP_APP_URL = "https://${APP_NAME_APP_URL}-master.${env.PATHFINDER_APP_URL}/"
```
Comment (asanchezr) [inline Jenkinsfile.dev]: Same here... should be `APP_URL`, not **APP_APP_URL**

File: Jenkinsfile.dev [hunk 4]
```diff
@@ -112,12 +124,68 @@ if(hasRepoChanged){
               type: "button",
               text: "View Build Logs",
               style:"danger",        
-              url: "${currentBuild.absoluteUrl}/console"
+              APP_URL: "${currentBuild.absoluteAPP_URL}/console"
```
Comment (asanchezr) [inline Jenkinsfile.dev]: Please fix all instances of `${currentBuild.absoluteAPP_URL}` -> `${currentBuild.absoluteUrl}`

File: Jenkinsfile.dev [hunk 5]
```diff
@@ -112,12 +124,68 @@ if(hasRepoChanged){
...
             ]
           ])
       }
     }
   }
+
+  podTemplate(label: 'bddstack', name: 'bddstack', serviceAccount: 'jenkins', cloud: 'openshift', containers: [
+    containerTemplate(
+      name: 'jnlp',
+      image: '172.50.0.2:5000/openshift/jenkins-slave-bddstack',
```
Comment (asanchezr) [inline Jenkinsfile.dev]: Shouldn't the image reference come from the ENV? i.e. `env.JENKINS_BDD_IMAGE`? 
Or at least remove specific IP addresses from configuration and replace with env vars; e.g. env.CLUSTER_IP_ADDRESS or similar...

File: Jenkinsfile.dev (hunk 1 above)
Comment (esune) [reply Jenkinsfile.dev]: Yes, good catch. I keep doing this search/replace fails... -.-

File: Jenkinsfile.dev (hunk 5 above)
Comment (esune) [reply Jenkinsfile.dev]: Let me ask for more info and get back to you on this as I copied the step over from GWells.

File: config/assets/production.js [hunk 6]
```diff
@@ -10,7 +10,7 @@ module.exports = {
         'public/assimilated/ng-table/dist/ng-table.min.css'
```
Comment (esune) [inline config/assets/production.js]: Please ignore this file, just some formatting changes that sneaked in.

**Comment by esune** [comment]
Hopefully undid all the search-and-replace fails. I also added the ZAP scan to the pipeline as apparently it should work out of the box and it would be good to have.

**Pull request merged**